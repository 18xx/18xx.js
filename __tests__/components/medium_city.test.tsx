import { List } from 'immutable';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Point from '../../src/point';

import MediumCity from '../../src/components/medium_city';

describe('MediumCity', () => {
  describe('render', () => {
    it('renders an svg', () => {
      const points: List<Point> = List([
        new Point(10, 20),
        new Point(40, 40),
      ]);

      const subject: React.ReactElement<MediumCity> = (
        <MediumCity points={points} />
      );
      expect(renderer.create(subject)).toMatchSnapshot();
    });
  });
});
