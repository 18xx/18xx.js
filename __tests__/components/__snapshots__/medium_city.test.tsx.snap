// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediumCity render renders an svg 1`] = `
<g>
  <g>
    <circle
      cx={10}
      cy={20}
      fill="none"
      r={20}
      stroke="black"
      strokeWidth={2}
    />
    <circle
      cx={10}
      cy={20}
      fill="black"
      r={10}
      stroke="black"
      strokeWidth={2}
    />
  </g>
  <g>
    <circle
      cx={40}
      cy={40}
      fill="none"
      r={20}
      stroke="black"
      strokeWidth={2}
    />
    <circle
      cx={40}
      cy={40}
      fill="black"
      r={10}
      stroke="black"
      strokeWidth={2}
    />
  </g>
</g>
`;
